subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -5661,9 +5662,19 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 		}
 	}
 #endif /* CONFIG_LWM2M_DTLS_SUPPORT */
+	if ((client_ctx->remote_addr).sa_family == AF_INET) {
+		addr_len = sizeof(struct sockaddr_in);
+	}
+	else if ((client_ctx->remote_addr).sa_family == AF_INET6) {
+		addr_len = sizeof(struct sockaddr_in6);
+	}
+	else {
+		lwm2m_engine_context_close(client_ctx);
+		return -EPROTONOSUPPORT;
+	}
 
 	if (connect(client_ctx->sock_fd, &client_ctx->remote_addr,
-		    NET_SOCKADDR_MAX_SIZE) < 0) {
+		addr_len) < 0) {
```

Comment:
Please keep the alignment.
```suggestion
		    addr_len) < 0) {
```